src/analytics/AnalyticTable.js

```diff
@@ -0,0 +1,144 @@
+      if (key !== 'Total') {
+        groups[key] = sortWithTotal(groups[key], secondary);
+        for (const subkey of Object.keys(groups[key])) {
+          if (!groups.Total[subkey]) {
+            groups.Total[subkey] = 0;
+          }
+          groups.Total[subkey] += groups[key][subkey];
+        }
+      }
+    }
+    return groups;
+  }, [cards, primary, secondary]);
+  const entryRenderer = useCallback(
+    (value) => (
+      <>
+        {valueRenderer(value ?? 0)}
+        <span className="percent">{valueRenderer(((value ?? 0) / primaryGroups.Total.Total) * 100)}%</span>
+      </>
+    ),
+    [primaryGroups],
+  );
+
+  const primaryLabels = useMemo(
+    () =>
+      getLabels(cards, primary)
+        .filter((label) => primaryGroups[label])
+        .concat(['Total']),
+    [cards, primary, primaryGroups],
+  );
+  const rows = useMemo(() => {
+    const secondaryGroups = sortIntoGroups(cards, secondary);
+    return getLabels(cards, secondary)
+      .filter((label) => secondaryGroups[label])
+      .concat(['Total'])
+      .map((secondaryLabel) => ({
+        secondaryLabel,
+        ...fromEntries(
+          primaryLabels.map((primaryLabel) => [primaryLabel, primaryGroups[primaryLabel][secondaryLabel] ?? 0]),
+        ),
+      }));
+  }, [cards, secondary, primaryLabels, primaryGroups]);
+  const columnProps = useMemo(
+    () => [
+      { key: 'secondaryLabel', title: '', heading: true, sortable: true },
+      ...primaryLabels.map((title) => ({ key: title, title, heading: false, sortable: true, renderFn: entryRenderer })),
+    ],
+    [entryRenderer, primaryLabels],
+  );
+
+  return (
+    <>
+      <Row>
+        <Col>
+          <h4 className="d-lg-block d-none">Table</h4>
+          <p>View card counts and percentages.</p>
+          <InputGroup className="mb-3">
+            <InputGroupAddon addonType="prepend">
+              <InputGroupText>Columns: </InputGroupText>
+            </InputGroupAddon>
+            <CustomInput type="select" value={primary} onChange={(event) => setPrimary(event.target.value)}>
+              {SORTS.map((item) => (
+                <option key={item} value={item}>
+                  {item}
+                </option>
+              ))}
+            </CustomInput>
+          </InputGroup>
+          <InputGroup className="mb-3">
+            <InputGroupAddon addonType="prepend">
+              <InputGroupText>Rows: </InputGroupText>
+            </InputGroupAddon>
+            <CustomInput type="select" value={secondary} onChange={(event) => setSecondary(event.target.value)}>
+              {SORTS.map((item) => (
+                <option key={item} value={item}>
+                  {item}
+                </option>
+              ))}
+            </CustomInput>
+          </InputGroup>
+        </Col>
+      </Row>
+      <AsfanDropdown cube={cube} defaultFormatId={defaultFormatId} setAsfans={setAsfans} />
+      <ErrorBoundary>
+        <SortableTable
```

Comment:
It feels a little weird visually that the leftmost column still displays the sortable arrows when no heading is specified for it (see below). I'd either remove them from this column, or add a heading to it (based on the value of `secondary`).

![Screenshot from 2021-01-15 13-42-13](https://user-images.githubusercontent.com/38463785/104728878-77356700-572f-11eb-98cf-2ae4597673ee.png)

Reply:
I'll just have it show the secondary sort key since I would like to keep it sortable.